balor/balor.ino

```diff
@@ -68,10 +76,25 @@ void driveCallback(const ackermann_msgs::AckermannDrive& drive_msg)
   debug_drive_pub.publish(&debug_drive_msg);
 }
 
+/**
+ * @brief get all the required params from the ROS param server
+ */
+void getRequiredParams()
+{
```

Comment:
Had to do the following `if()` check, as the getParam would not reliably work otherwise, as explained in [this Wiki](https://github.com/kittcar/kitt_platform/wiki/ros_serial-limitations)

Reply:
Let's change this so that `getRequiredParams` returns `false` and the checks below are `if !` which return `false`.
Then `setup` sets a global flag, which `loop` checks.
If this check returns `false` then only the heartbeat is allowed to be published.
We should then publish an error state.